x-pack/plugins/lens/public/indexpattern_datasource/operations/definitions/filters/filters.tsx

```diff
@@ -0,0 +1,282 @@
+        setFilters(items);
+      }
+    };
+
+    const setFilters = (newFilters: Filter[]) =>
+      setState(
+        updateColumnParam({
+          state,
+          layerId,
+          currentColumn,
+          paramName: 'filters',
+          value: newFilters,
+        })
+      );
+
+    return (
+      <EuiForm>
+        <EuiFormRow
+          label={i18n.translate('xpack.lens.indexPattern.filters.queries', {
+            defaultMessage: 'Queries',
+          })}
+        >
+          <div>
+            <EuiDragDropContext onDragEnd={onDragEnd}>
+              <EuiDroppable droppableId="FILTERS_DROPPABLE_AREA" spacing="s">
+                {filters?.map((filter: Filter, idx: number) => {
+                  const { input, label } = filter;
+                  const id = `${JSON.stringify(input.query)}_${label}`;
+                  return (
+                    <EuiDraggable
+                      spacing="m"
+                      key={id}
+                      index={idx}
+                      draggableId={id}
+                      customDragHandle={true}
+                    >
+                      {(provided) => (
+                        <EuiPanel paddingSize="none">
+                          <EuiFlexGroup gutterSize="xs" alignItems="center" responsive={false}>
+                            <EuiFlexItem grow={false}>
+                              <div {...provided.dragHandleProps} className="lnsLayerPanel__dndGrab">
+                                <EuiIcon
+                                  type="grab"
+                                  aria-label={i18n.translate(
+                                    'xpack.lens.indexPattern.filters.grabIcon',
+                                    {
+                                      defaultMessage: 'Grab icon',
+                                    }
+                                  )}
+                                />
+                              </div>
+                            </EuiFlexItem>
+                            <EuiFlexItem
+                              grow={true}
+                              data-test-subj="indexPattern-filters-existingFilterContainer"
+                            >
+                              <FilterPopover
+                                indexPattern={indexPattern}
+                                filter={filter}
+                                Button={({ onClick }: { onClick: MouseEventHandler }) => (
+                                  <EuiLink
+                                    color="text"
+                                    onClick={onClick}
+                                    className="lnsLayerPanel__filterLink"
+                                    data-test-subj="indexPattern-filters-existingFilterTrigger"
+                                  >
+                                    <EuiText size="s" textAlign="left">
+                                      {label ? label : input.query}
```

Comment:
Labeling using `input.query` will show `[object Object]`, so I think we want something different as a fallback.